guvnor-ala/guvnor-ala-openshift/guvnor-ala-openshift-provider/src/main/java/org/guvnor/ala/openshift/executor/OpenShiftProviderConfigExecutor.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.guvnor.ala.openshift.executor;
+
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.guvnor.ala.config.Config;
+import org.guvnor.ala.config.ProviderConfig;
+import org.guvnor.ala.openshift.config.OpenShiftProviderConfig;
+import org.guvnor.ala.openshift.model.OpenShiftProvider;
+import org.guvnor.ala.openshift.model.OpenShiftProviderImpl;
+import org.guvnor.ala.pipeline.FunctionConfigExecutor;
+import org.guvnor.ala.registry.RuntimeRegistry;
+import org.guvnor.ala.runtime.providers.ProviderBuilder;
+import org.guvnor.ala.runtime.providers.ProviderDestroyer;
+import org.guvnor.ala.runtime.providers.ProviderId;
+
+public class OpenShiftProviderConfigExecutor implements ProviderBuilder<OpenShiftProviderConfig, OpenShiftProvider>, ProviderDestroyer, FunctionConfigExecutor<OpenShiftProviderConfig, OpenShiftProvider> {
+
+    private RuntimeRegistry runtimeRegistry;
+
+    @Inject
+    public OpenShiftProviderConfigExecutor(final RuntimeRegistry runtimeRegistry) {
+        this.runtimeRegistry = runtimeRegistry;
+    }
+
+    @Override
+    public Optional<OpenShiftProvider> apply(final OpenShiftProviderConfig openshiftProviderConfig) {
+        final OpenShiftProviderImpl provider = new OpenShiftProviderImpl(openshiftProviderConfig.getName(), openshiftProviderConfig);
```

Comment:
Hello @errantepiphany quick note. This piece here is fine, but seems it was based on old code as reference. This is of course not your fault since we have a bottleneck right now with the PRs, etc.

Can you please take a look at this newer version below and as reference and refactor your version?
The change is not big, basically the old code always registers the Provider, but in general when the provider is registered we just use it, if not then we register it.

See:

https://github.com/wmedvede/guvnor/blob/c7b7c43ae3c07d8720addab564ff057bc30145aa/guvnor-ala/guvnor-ala-wildfly/guvnor-ala-wildfly-provider/src/main/java/org/guvnor/ala/wildfly/executor/WildflyProviderConfigExecutor.java#L49-L49
 
Thanks!